src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsMigrationService.kt

```diff
@@ -69,22 +70,11 @@ class NonAssociationsMigrationService(
     with(context.body) {
       log.info("attempting to migrate $this")
 
-      if (isNotPrimaryNonAssociation()) {
-        telemetryClient.trackEvent(
-          "non-association-migration-non-primary-skipped",
-          mapOf(
-            "firstOffenderNo" to offenderNo1,
-            "secondOffenderNo" to offenderNo2,
-            "migrationId" to context.migrationId,
-          ),
-        )
-        return
-      }
-
       // Determine all valid non-associations for this offender pair
       val nonAssociationPairs = nomisApiService.getNonAssociations(offenderNo1, offenderNo2)
+      val latestOpenTypeSequence = nonAssociationPairs.findLatestOpenTypeSequence()
```

Comment:
Added initial call here to loop to determine latestOpen NonAssociation type sequence (was too messy inside loop)